cardano_node_tests/tests/test_delegation.py

```diff
@@ -631,12 +651,16 @@ def test_addr_delegation_deregistration(
 
         # create stake address registration cert
         stake_addr_reg_cert_file = cluster.g_stake_address.gen_stake_addr_registration_cert(
-            addr_name=f"{temp_template}_addr0", stake_vkey_file=user_registered.stake.vkey_file
+            addr_name=f"{temp_template}_addr0",
+            stake_vkey_file=user_registered.stake.vkey_file if stake_cert == "vkey_file" else None,
+            stake_address=user_registered.stake.address if stake_cert == "stake_address" else None,
```

Comment:
Optional: evaluating the conditions just once at the beginning of the test function would be more readable IMO

```python
stake_vkey_file = user_registered.stake.vkey_file if stake_cert == "vkey_file" else None
stake_address = user_registered.stake.address if stake_cert == "stake_address" else None
```

and then using `stake_vkey_file` and `stake_address` where needed.